src/Avalonia.FreeDesktop/LinuxMountedVolumeInfoListener.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+using Avalonia.Controls.Platform;
+using Tmds.DBus;
+using System.Reactive.Disposables;
+
+namespace Avalonia.FreeDesktop.Dbus
+{
+    public partial class LinuxMountedVolumeInfoProvider
+    {
+        private class LinuxMountedVolumeInfoListener : IDisposable
+        {
+            private CompositeDisposable _disposables;
+            private readonly Connection _sysDbus;
+            private readonly IObjectManager _udisk2Manager;
+            private readonly ObservableCollection<MountedVolumeInfo> _targetObs;
+            private bool disposedValue = false;
+
+            public LinuxMountedVolumeInfoListener(ref ObservableCollection<MountedVolumeInfo> target)
+            {
+                this._sysDbus = Connection.System;
+                this._udisk2Manager = _sysDbus.CreateProxy<IObjectManager>("org.freedesktop.UDisks2", "/org/freedesktop/UDisks2");
```

Comment:
DBus and udisks2 aren't guaranteed to be present in the system. We need to fall back to conservative way of obtaining the list of mount points.